.github/workflows/test.yaml

```diff
@@ -11,50 +10,129 @@ on:
+              - 'nbsite/**'
+              - 'pixi.toml'
+              - 'pyproject.toml'
+              - '.github/workflows/test.yaml'
+              - 'scripts/nbsite_*.py'
+      - name: Set matrix option
+        run: |
+          if [[ '${{ github.event_name }}' == 'workflow_dispatch' ]]; then
+            OPTION=${{ github.event.inputs.target }}
+          elif [[ '${{ github.event_name }}' == 'schedule' ]]; then
+            OPTION="full"
+          elif [[ '${{ github.event_name }}' == 'push' && '${{ github.ref_type }}' == 'tag' ]]; then
+            OPTION="full"
+          else
+            OPTION="default"
+          fi
+          echo "MATRIX_OPTION=$OPTION" >> $GITHUB_ENV
+      - name: Set test matrix with 'default' option
+        if: env.MATRIX_OPTION == 'default'
+        run: |
+          MATRIX=$(jq -nsc '{
+              "os": ["ubuntu-latest", "macos-latest", "windows-latest"],
+              "environment": ["test-39", "test-312"],
+            }')
+          echo "MATRIX=$MATRIX" >> $GITHUB_ENV
+      - name: Set test matrix with 'full' option
+        if: env.MATRIX_OPTION == 'full'
+        run: |
+          MATRIX=$(jq -nsc '{
+              "os": ["ubuntu-latest", "macos-latest", "windows-latest"],
+              "environment": ["test-39", "test-310", "test-311", "test-312", "test-313"],
+            }')
+          echo "MATRIX=$MATRIX" >> $GITHUB_ENV
+
+  pixi_lock:
+    name: Pixi lock
+    runs-on: ubuntu-latest
+    steps:
+      - uses: holoviz-dev/holoviz_tasks/pixi_lock@v0
+        with:
+          cache: ${{ github.event.inputs.cache == 'true' || github.event.inputs.cache == '' }}
+
+  unit_test_suite:
+    name: unit:${{ matrix.environment }}:${{ matrix.os }}
+    needs: [pre_commit, setup, pixi_lock]
     runs-on: ${{ matrix.os }}
+    if: needs.setup.outputs.code_change == 'true'
     strategy:
       fail-fast: false
-      matrix:
-        os: ['ubuntu-latest']
-        python-version: ['3.9', '3.10', '3.11', '3.12']
-    timeout-minutes: 30
-    defaults:
-      run:
-        shell: bash -l {0}
-    env:
-      PYTHON_VERSION: ${{ matrix.python-version }}
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      matrix: ${{ fromJson(needs.setup.outputs.matrix) }}
+    timeout-minutes: 60
     steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: "100"
-      - uses: actions/setup-python@v5
+      - uses: holoviz-dev/holoviz_tasks/pixi_install@v0
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Fetch
-        run: git fetch --prune --tags
-      - name: env setup
+          environments: ${{ matrix.environment }}
+      - name: Test Unit
         run: |
-          set -xe
-          python -VV
-          python -m pip install --upgrade pip
-          python -m pip install -e ".[tests]"
-      - name: pip list
-        run: pip list
-      - name: tests
-        run: pytest nbsite
+          pixi run -e ${{ matrix.environment }} test-unit
```

Comment:
Can we add `$COV` here?

Reply:
OK. Although it seemed you wanted it removed before.